python/pyiceberg/expressions/literals.py

```diff
@@ -439,6 +439,9 @@ def _(self, type_var: DecimalType) -> Optional[Literal[Decimal]]:
         else:
             return None
 
+    def __repr__(self) -> str:
+        return f"StringLiteral({repr(self.value)})"
```

Comment:
I had to add this one, otherwise, it would produce `StringLiteral(foo)` which isn't valid Python (missing quotes around the string.